components/script/dom/globalscope.rs

```diff
@@ -1619,32 +1589,28 @@ impl GlobalScope {
     /// Get bytes from a non-sliced blob
     fn get_blob_bytes_non_sliced(&self, blob_id: &BlobId) -> Result<Vec<u8>, ()> {
         let blob_state = self.blob_state.borrow();
-        if let BlobState::Managed(blobs_map) = &*blob_state {
-            let blob_info = blobs_map
-                .get(blob_id)
-                .expect("get_blob_bytes_non_sliced called for a unknown blob.");
-            match blob_info.blob_impl.blob_data() {
-                BlobData::File(ref f) => {
-                    let (buffer, is_new_buffer) = match f.get_cache() {
-                        Some(bytes) => (bytes, false),
-                        None => {
-                            let bytes = self.read_file(f.get_id())?;
-                            (bytes, true)
-                        },
-                    };
+        let blob_info = blob_state
+            .get(blob_id)
+            .expect("get_blob_bytes_non_sliced called for a unknown blob.");
+        match blob_info.blob_impl.blob_data() {
+            BlobData::File(ref f) => {
+                let (buffer, is_new_buffer) = match f.get_cache() {
+                    Some(bytes) => (bytes, false),
+                    None => {
+                        let bytes = self.read_file(f.get_id())?;
+                        (bytes, true)
+                    },
+                };
 
-                    // Cache
-                    if is_new_buffer {
-                        f.cache_bytes(buffer.clone());
-                    }
+                // Cache
+                if is_new_buffer {
+                    f.cache_bytes(buffer.clone());
+                }
 
-                    Ok(buffer)
-                },
-                BlobData::Memory(ref s) => Ok(s.clone()),
-                BlobData::Sliced(_, _) => panic!("This blob doesn't have a parent."),
-            }
-        } else {
-            panic!("get_blob_bytes_non_sliced called on a global not managing any blobs.");
```

Comment:
I also don't like that we're now not making a difference between "unknown blob while the global is managing", and "unknown blob and the global is not managing". Those are two different failures in the current workflow, whereas now we merge them into one. They're supposed to be unreachable, but if some future bug where to reach them, it would make debugging far easier to keep them distinct. 

I think you can add a debug about whether the map is empty to do something similar. 

Once again it shows that abstraction has a usecase in clarifying the properties of the workflow involved.